openupgrade_scripts/scripts/loyalty/16.0.1.0/pre-migration.py

```diff
@@ -0,0 +1,806 @@
+            write_date
+        )
+        SELECT
+            lp.active,
+            gc.program_id,
+            gc.company_id,
+            '{"en_US": "Gift Card"}',
+            'discount',
+            1,
+            'per_point',
+            'order',
+            sol.product_id,
+            false,
+            1,
+            1,
+            gc.create_uid,
+            gc.write_uid,
+            '[]',
+            gc.create_date,
+            gc.write_date
+        FROM gift_card AS gc
+        JOIN loyalty_program AS lp ON gc.program_id = lp.id
+        JOIN sale_order_line AS sol ON sol.gift_card_id = gc.id
+        """,
+    )
+    # Create records in loyalty_rule based on gift_card data
+    openupgrade.logged_query(
+        env.cr,
+        """
+        INSERT INTO loyalty_rrule (
+            program_id,
+            company_id,
+            minimun_qty,
+            create_uid,
+            write_uid,
+            product_domain,
+            reward_point_mode,
+            minimun_amount_tax_mode,
+            mode,
+            active,
+            reward_point_split,
+            create_date,
+            write_date,
+            reward_point_amount
+        )
+        SELECT
+            gc.program_id,
+            gc.company_id,
+            1,
+            gc.create_uid,
+            gc.write_uid,
+            '[]',
+            'money',
+            'incl',
+            'auto',
+            lp.active,
+            true,
+            gc.create_date,
+            gc.write_date,
+            1
+        FROM gift_card AS gc
+        JOIN loyalty_program AS lp ON gc.program_id = lp.id
+        """,
+    )
+    # If the card has been used, it calculates the new points value by subtracting the
+    # sum of the absolute price_unit values of the order lines linked to a gift_card
+    # from the initial value of the loyalty card (loyalty_card.initial_amount). In case
+    # there is no order line linked to the gift card, the initial value of the loyalty
+    # card shall be maintained.
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE loyalty_card lc
+        SET points =
+            CASE
+                WHEN (
+                    SELECT SUM(ABS(sol.price_unit))
+                    FROM sale_order_line sol
+                    JOIN gift_card gc ON gc.id = sol.gift_card_id
+                    WHERE gc.program_id = lc.program_id
+                ) IS NOT NULL
+                THEN lc.initial_amount - (
+                    SELECT SUM(ABS(sol.price_unit))
+                    FROM sale_order_line sol
+                    JOIN gift_card gc ON gc.id = sol.gift_card_id
+                    WHERE gc.program_id = lc.program_id
+                )
+                ELSE lc.points
+            END
+        FROM gift_card gc
+        WHERE lc.program_id = gc.program_id
+        """,
+    )
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.rename_fields(env.cr, _field_renames)
```

Comment:
Please change to pass only env, on current version of lib raise error https://github.com/OCA/openupgradelib/blob/master/openupgradelib/openupgrade.py#L634
```suggestion
    openupgrade.rename_fields(env, _field_renames)
```

Reply:
thanks :)